src/components/ConsoleIO/ConsoleIO.tsx

```diff
@@ -0,0 +1,108 @@
+import { CONSOLE_DEFAULTS } from './ConsoleIO.constants';
+import { useConsoleIO } from './ConsoleIO.hook';
+import {
+  ConsoleContainer,
+  Frame,
+  TopTabs,
+  StyledTab,
+  ConsoleContent,
+  ScrollableTextField,
+} from './ConsoleIO.styles';
+
+import type { ConsoleIOProps } from './ConsoleIO.types';
+
+export const ConsoleIO: React.FC<ConsoleIOProps> = ({
+  sampleValue = '',
+  customValue = '',
+  onSampleChange,
+  onCustomChange,
+  defaultTab = 'sample',
+  height = CONSOLE_DEFAULTS.HEIGHT,
+  width,
+  'data-testid': dataTestId,
+}) => {
+  const {
+    activeTab,
+    sampleInput,
+    customInput,
+    handleTabChange,
+    handleSampleChange,
+    handleCustomChange,
+  } = useConsoleIO({
+    sampleValue,
+    customValue,
+    defaultTab,
+    onSampleChange,
+    onCustomChange,
+  });
+
+  const outerStyle: React.CSSProperties = {
+    width: typeof width === 'number' ? `${width}px` : (width ?? '100%'),
+    maxWidth: `${CONSOLE_DEFAULTS.WIDTH}px`,
+    height: typeof height === 'number' ? `${height}px` : height,
+  };
+
+  return (
+    <ConsoleContainer data-testid={dataTestId ?? 'consoleio-root'} style={outerStyle}>
```

Comment:
The component applies inline styles via the `style` prop (outerStyle) which includes px string values computed from constants. The project guideline requires using `.styles.ts` or MUI `sx` for styling and avoiding inline px where possible.

**Sugerencia:**
```typescript
Move width/height/maxWidth logic into the styled component or provide an `sx` prop: `<ConsoleContainer sx={{ width, maxWidth: ... }} />` or create a styled wrapper that accepts props for width/height. Use theme-aware spacing instead of raw px strings.
```